server/status/runtime.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package status
+
+import (
+	"fmt"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util/hlc"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// RuntimeStatRecorder is used to periodically persist useful runtime statistics
+// as time series data. "Runtime statistics" include OS-level statistics (such as
+// memory and CPU usage) and Go runtime statistics (e.g. count of Goroutines).
+type RuntimeStatRecorder struct {
```

Comment:
Add a TODO to use a metrics package for assembling the data (not the ad-hoc stuff I've been doing in the code below). I'm currently looking at options.